modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -173,12 +172,20 @@ public CompletableFuture<Void> notifyWatches(List<Entry> updatedEntries, HybridT
                                 CompletableFuture<Void> notifyUpdateRevisionFuture = notifyUpdateRevisionListeners(newRevision);
 
                                 CompletableFuture<Void> notificationFuture = allOf(notifyWatchesFuture, notifyUpdateRevisionFuture)
-                                        .thenAcceptAsync(
-                                                unused -> invokeOnRevisionCallback(newRevision, time),
+                                        .thenRunAsync(
+                                                () -> invokeOnRevisionCallback(newRevision, time),
                                                 watchExecutor
                                         );
 
-                                notificationFuture.whenComplete((unused, e) -> maybeLogLongProcessing(updatedEntries, startTimeNanos));
+                                notificationFuture.whenComplete((unused, e) -> {
+                                    maybeLogLongProcessing(updatedEntries, startTimeNanos);
+
+                                    if (e != null) {
+                                        LOG.error("Error occurred when notifying watches", e);
+
+                                        throw new CompletionException(e);
```

Comment:
This `throw` is redundant, it won't be propagated anywhere

Reply:
That's right, I feel stupid now